Moonscraper Chart Editor/Assets/Scripts/Game/Misc/SongValidate.cs

```diff
@@ -1,242 +1,262 @@
-            var timeSignatures = song.timeSignatures;
-            for (int tsIndex = 1; tsIndex < timeSignatures.Count; ++tsIndex)
-            {
-                TimeSignature previousTs = timeSignatures[tsIndex - 1];
-                TimeSignature.MeasureInfo measureInfo = previousTs.GetMeasureInfo();
-
-                TimeSignature tsToTest = timeSignatures[tsIndex];
-                uint deltaTick = tsToTest.tick - previousTs.tick;
-
-                var measureLine = measureInfo.measureLine;
-                if (((float)deltaTick % measureLine.tickGap) != 0)      // Doesn't line up on a measure
-                {
-                    hasErrorsLocal |= true;
-                    sb.AppendFormat("\tFound misaligned Time Signature at time {1}, position {0}. Time signatures must be aligned to the measure set by the previous time signature.\n", 
-                        tsToTest.tick
-                        , PrintObjectTime(tsToTest.time)
-                    );
-                }
-            }
-        }
-
-        // If we have no starpower but more than 1 solo section then CH will interpret this as an RB1 style midi, and misinterpret the solo markers as starpower
-        if (validationParams.checkMidiIssues)
-        {
-            foreach (Song.Instrument instrument in EnumX<Song.Instrument>.Values)
-            {
-                if (instrument == Song.Instrument.Unrecognised)
-                    continue;
-
-                foreach (Song.Difficulty difficulty in EnumX<Song.Difficulty>.Values)
-                {
-                    Chart chart = song.GetChart(instrument, difficulty);
-
-                    if (chart.starPower.Count <= 0)
-                    {
-                        // Check for solo markers
-                        int soloMarkerCount = 0;
-                        foreach (ChartEvent ce in chart.events)
-                        {
-                            if (ce.eventName == MoonscraperChartEditor.Song.IO.MidIOHelper.SoloEventText)
-                            {
-                                ++soloMarkerCount;
-
-                                if (soloMarkerCount > 1)
-                                {
-                                    hasErrorsLocal |= true;
-                                    sb.AppendFormat("\tTrack {0} has no starpower and more than 1 solo section. If exported to the midi format, Clone Hero will interpret this chart as an older style midi, and will misinterpret solo markers as starpower.\n", instrument);
-
-                                    goto NewInstrument;
-                                }
-                            }
-                        }
-                    }
-                }
-
-            NewInstrument:;
-            }
-        }
-
-        if (!hasErrorsLocal)
-        {
-            sb.AppendLine("\tNo errors detected");
-        }
-
-        hasErrors |= hasErrorsLocal;
-
-        return sb.ToString();
-    }
-}
+// Copyright (c) 2016-2020 Alexander Ong

```

Comment:
Something has gone wrong with the line endings on this file. Make sure your git has autocrlf set to true, then try running `git add --renormalize .` and recommitting.